activesupport/lib/active_support/cache.rb

```diff
@@ -186,35 +186,44 @@ class << self
           private_constant :DEFAULT_POOL_OPTIONS
 
           def retrieve_pool_options(options)
-            unless options.key?(:pool) || options.key?(:pool_size) || options.key?(:pool_timeout)
-              options[:pool] = true
-            end
+            if options.key?(:pool)
+              pool_options = options.delete(:pool)
+            elsif options.key?(:pool_size) || options.key?(:pool_timeout)
+              pool_options = {}
+
+              if options.key?(:pool_size)
+                ActiveSupport.deprecator.warn(<<~MSG)
+                  Using :pool_size is deprecated and will be removed in Rails 7.2.
+                  Use `pool: { size: #{options[:pool_size].inspect} }` instead.
+                MSG
+                pool_options[:size] = options.delete(:pool_size)
+              end
 
-            if (pool_options = options.delete(:pool))
-              if Hash === pool_options
-                DEFAULT_POOL_OPTIONS.merge(pool_options)
-              else
-                DEFAULT_POOL_OPTIONS
+              if options.key?(:pool_timeout)
+                ActiveSupport.deprecator.warn(<<~MSG)
+                  Using :pool_timeout is deprecated and will be removed in Rails 7.2.
+                  Use `pool: { timeout: #{options[:pool_timeout].inspect} }` instead.
+                MSG
+                pool_options[:timeout] = options.delete(:pool_timeout)
               end
             else
-              {}.tap do |pool_options|
-                if options[:pool_size]
-                  ActiveSupport.deprecator.warn(<<~MSG)
-                    Using :pool_size is deprecated and will be removed in Rails 7.2.
-                    Use `pool: { size: #{options[:pool_size].inspect} }` instead.
-                  MSG
-                  pool_options[:size] = options.delete(:pool_size)
-                end
-
-                if options[:pool_timeout]
-                  ActiveSupport.deprecator.warn(<<~MSG)
-                    Using :pool_timeout is deprecated and will be removed in Rails 7.2.
-                    Use `pool: { timeout: #{options[:pool_timeout].inspect} }` instead.
-                  MSG
-                  pool_options[:timeout] = options.delete(:pool_timeout)
-                end
-              end
+              pool_options = true
```

Comment:
@casperisfine it seems that this change the default behavior of
`ActiveSupport::Cache::Store.retrieve_pool_options`
and causes [ActiveSupport::Cache::RedisCacheStore#initialize](https://github.com/rails/rails/blob/600a052c8cf6d355bd2d3f867cecce7cefdb1894/activesupport/lib/active_support/cache/redis_cache_store.rb#L148-L152) to use a `::ConnectionPool ` when it did not before
 
```ruby
if pool_options = self.class.send(:retrieve_pool_options, redis_options)
  @redis = ::ConnectionPool.new(pool_options) { self.class.build_redis(**redis_options) }
else
  @redis = self.class.build_redis(**redis_options)
end
```  

WDYT?

Reply:
It's on purpose, 99% of the time not using a pool is unintentional.

Worst case if you only need a single connection, `ConnectionPool` lazily create them so it's fine.

Is this causing problem to you?

Reply:
When we upgraded to Rails 7.1, our use of `scan_each`, [here](https://www.rubydoc.info/gems/redis/Redis/Commands/Keys#scan_each-instance_method), broke because `scan_each` is not implemented by the `::ConnectionPool` that `ActiveSupport::Cache::RedisCacheStore#redis` is now returning. 
No big deal.